R/redundant_ifelse_linter.R

```diff
@@ -46,17 +46,17 @@ redundant_ifelse_linter <- function(allow10 = FALSE) {
     if (!allow10) {
       num_expr <- xml2::xml_find_all(xml, num_xpath)
       matched_call <- xp_call_name(num_expr)
-      # [1] call; [2] logical condiditon
+      # [1] call; [2] logical condition
       first_arg <- xml2::xml_find_chr(num_expr, "string(expr[3]/NUM_CONST)")
       second_arg <- xml2::xml_find_chr(num_expr, "string(expr[4]/NUM_CONST)")
-      replacement <- ifelse(
-        first_arg %in% c("0", "1") | second_arg %in% c("0", "1"),
-        "as.numeric",
-        "as.integer"
-      )
+      is_numeric_01 <- first_arg %in% c("0", "1") | second_arg %in% c("0", "1")
+      replacement_is_numeric <- ifelse(is_numeric_01, "as.numeric", "as.integer")
```

Comment:
not: better naming. current reads like it's a boolean variable. maybe "coercion_function"?